python/cudf/cudf/core/indexed_frame.py

```diff
@@ -2090,26 +2089,19 @@ def duplicated(self, subset=None, keep="first"):
         subset = self._preprocess_subset(subset)
 
         if isinstance(self, cudf.Series):
-            df = self.to_frame(name="None")
-            subset = ["None"]
+            columns = [self._column]
         else:
-            df = self.copy(deep=False)
-        df._data["index"] = cudf.core.column.arange(
-            0, len(self), dtype=size_type_dtype
+            columns = [self._data[n] for n in subset]
+        distinct = libcudf.stream_compaction.distinct_indices(
+            columns, keep=keep
         )
-
-        new_df = df.drop_duplicates(subset=subset, keep=keep)
-        idx = df.merge(new_df, how="inner")["index"]
-        s = cudf.Series._from_data(
-            {
-                None: cudf.core.column.full(
-                    size=len(self), fill_value=True, dtype="bool"
-                )
-            },
-            index=self.index,
+        (result,) = libcudf.copying.scatter(
+            [full(len(distinct), False, dtype=bool)],
+            distinct,
+            [full(len(self), True, dtype=bool)],
```

Comment:
Is it possible to scatter (or "set") a **scalar** `True` value over a column of `False` instead of scattering from column-of-true to column-of-false?

Reply:
I can probably scatter a scalar False to a column of True. To scatter a True to a column of False would require one invert the mask first

Reply:
I didn't check which direction it's supposed to go, I just wanted to avoid allocating two full columns. Please verify my claim. :)

Reply:
Implemented via scatter-scalar.